main.py

```
@@ -53,219 +87,1126 @@
+            status_code=503,
+            detail="Vertex AI integration not available"
+        )
+    
+    try:
+        config = get_vertex_deployment_config()
+        return config
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Failed to get Vertex AI config: {str(e)}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)